### DISCUSSION TITLE
Where do the input variables for "/defaults/main.yaml" come from?

### DISCUSSION THREAD

**Discussion by andsouth44** [aristanetworks/avd#3047] [Q&A] [answered]
Example:
In 
~/.ansible/collections/ansible_collections/arista/avd/roles/eos_designs/defaults/main.yaml

On line 3 I see:

root_dir: '{{ inventory_dir }}'

Where does 'inventory_dir' come from?

**Comment by carlbuchmann** [comment, marked as answer]
@andsouth44 `inventory_dir` is an ansible magic variable: https://docs.ansible.com/ansible/latest/reference_appendices/special_variables.html#term-inventory_dir

**Reply by andsouth44** [reply]
Thanks!
